ktor-http/common/src/io/ktor/http/HttpHeaderValueParser.kt

```diff
@@ -155,36 +155,43 @@ private fun parseHeaderValueParameter(text: String, start: Int, parameters: Lazy
 
 
 private fun parseHeaderValueParameterValue(value: String, start: Int): Pair<Int, String> {
-    var pos = start
-    while (pos <= value.lastIndex) {
-        when (value[pos]) {
-            '"' -> return parseHeaderValueParameterValueQuoted(value, pos + 1)
-            ';', ',' -> return pos to value.subtrim(start, pos)
-            else -> pos++
+    if (value.length == start) {
+        return start to ""
+    }
+
+    var position = start
+    if (value[start] == '"') {
+        return parseHeaderValueParameterValueQuoted(value, position + 1)
+    }
+
+    while (position <= value.lastIndex) {
+        when (value[position]) {
+            ';', ',' -> return position to value.subtrim(start, position)
+            else -> position++
         }
     }
-    return pos to value.subtrim(start, pos)
+    return position to value.subtrim(start, position)
 }
 
 private fun parseHeaderValueParameterValueQuoted(value: String, start: Int): Pair<Int, String> {
     var pos = start
     val sb = StringBuilder()
     while (pos <= value.lastIndex) {
-        val c = value[pos]
-        when (c) {
+        val current = value[pos]
```

Comment:
Nit: maybe `currentSymbol` or any shorter version of it would be a better name here?

Reply:
Thanks, will change